theforeman.org/pipelines/test/rpm_copr_packaging.groovy

```diff
@@ -0,0 +1,170 @@
+                            )
+
+                            compare_release = sh(
+                              script: "rpmdev-vercmp ${old_release} ${new_release}",
+                              returnStatus: true,
+                              label: "rpmdev-vercmp"
+                            )
+
+                            compare_new_to_one = sh(
+                              script: "rpmdev-vercmp 1 ${new_release}",
+                              returnStatus: true,
+                              label: "rpmdev-vercmp"
+                            )
+
+                            if (compare_version != VERCMP_EQUAL && (compare_new_to_one == VERCMP_OLDER || compare_new_to_one == VERCMP_EQUAL)) {
+                                echo "New version and release is reset to 1 for ${package_name}"
+                            } else if (compare_version != VERCMP_EQUAL && compare_new_to_one == VERCMP_NEWER) {
+                                // new version, but release was not reset
+                                sh """
+                                    echo 'Version updated but release was not reset back to 1 for ${package_name}"'
+                                    exit 1
+                                """
+                            } else if (compare_version == VERCMP_EQUAL && compare_release == VERCMP_NEWER) {
+                                echo "Version remained the same and release is reset to 1 for ${package_name}"
+                            } else {
+                                sh """
+                                    echo 'Version or release needs updating for ${package_name}"'
+                                    exit 1
+                                """
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        stage('Scratch Build Packages') {
+            when {
+                expression { packages_to_build }
+            }
+            steps {
+
+                withCredentials([file(credentialsId: 'theforeman-bot-copr', variable: 'copr_config')]) {
+                    obal(
+                        action: "scratch",
+                        extraVars: [
+                            'build_package_build_system': 'copr',
+                            'build_package_download_rpms': 'True',
+                            'build_package_archive_build_ids': 'True',
+                            'build_package_copr_config': copr_config,
+                        ],
+                        packages: packages_to_build
+                    )
+                }
+
+            }
+        }
+
+        stage('Repoclosure') {
+            when {
+                expression { packages_to_build }
+            }
+            steps {
+
+                obal(action: "repoclosure", packages: packages_to_build)
+
+            }
+        }
+    }
+
+    post {
+        always {
+            status_copr_links(ghprbGhRepository.split('/')[1])
```

Comment:
Based on https://github.com/theforeman/foreman-packaging/pull/9290 this did not appear to work. I don't think that should be a blocker, but I will try to get it working by the time the other changes get merged.

Reply:
This works now.